### PR TITLE
DB-6012: avoid deleting a nonexist snapshot

### PR DESCRIPTION
**Pull request by jyuanca**
Add a hbase snapshot to a list after it is taken successfully. The list can be used to delete the snapshot if the txn needs to roll back.

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java [hunk 1]
```diff
@@ -1190,6 +1189,7 @@ private static void snapshot(ResultSet rs, String snapshotName, String schemaNam
                     new SnapshotDescriptor(snapshotName, schemaName, objectName, conglomerateNumber,creationTime, null);
             admin.snapshot(sname, "splice:" + conglomerateNumber);
             dd.addSnapshot(descriptor, tc);
+            snapshotList.add(sname);
```
Comment (snowspeeder) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java]: not sure how it helps, list add() is only moved down in the same loop, there is no condition

Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java]: My understanding is that if "admin.snapshot()" or "dd.addSnapshot()" fails, we should abort/return early and we won't reach the line "snapshotList.add(sname)". Later on, deleteSnapshots() in the below code won't delete a snapshot that fails to be created:

        try {
            dd.startWriting(lcc);

            ResultSet rs = getResultSet(sql1, schemaName, tableName);
            snapshot(rs, snapshotName, schemaName, dd, tc, snapshotList);

            rs = getResultSet(sql2, schemaName, tableName);
            snapshot(rs, snapshotName, schemaName, dd, tc, snapshotList);
        }
        catch (Exception e)
        {
            deleteSnapshots(snapshotList);
            throw e;
        }

**Comment by snowspeeder** [comment]
Makes sense. Thanks!

File: splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java (hunk 1 above)
Comment (jyuanca) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java]: Yes

**Pull request merged**